src/main/java/mate/academy/controller/BookController.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.controller;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.BookDto;
+import mate.academy.dto.CreateBookRequestDto;
+import mate.academy.model.Book;
+import mate.academy.service.BookService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.findBookById(id);
+    }
+
+    @PostMapping
+    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+        return bookService.save(bookDto);
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteById(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public void updateBook(@PathVariable Long id, @RequestBody Book book) {
+        bookService.updateBook(id, book);
```

Comment:
The method 'updateBook()' in 'BookService' should accept a 'Book' object, but it is unclear if the 'Book' object is properly constructed from the request body. Consider validating and mapping the request body to a 'Book' object.